itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestHiveQueryResultSet.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hive.jdbc;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Properties;
+
+import org.junit.Test;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Schema;
+import org.apache.hive.service.cli.RowSet;
+import org.apache.hive.service.cli.RowSetFactory;
+import org.apache.hive.service.cli.TableSchema;
+import org.apache.hive.service.cli.thrift.EmbeddedThriftBinaryCLIService;
+import org.apache.hive.service.rpc.thrift.TFetchResultsReq;
+import org.apache.hive.service.rpc.thrift.TFetchResultsResp;
+import org.apache.hive.service.rpc.thrift.TRowSet;
+import org.apache.hive.service.rpc.thrift.TStatus;
+import org.apache.hive.service.rpc.thrift.TStatusCode;
+import org.apache.thrift.TException;
+
+import static org.apache.hive.jdbc.EmbeddedCLIServicePortal.EMBEDDED_CLISERVICE;
+import static org.apache.hive.jdbc.Utils.URL_PREFIX;
+import static org.apache.hive.service.rpc.thrift.TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V10;
+import static org.junit.Assert.assertEquals;
+
+public class TestHiveQueryResultSet {
+  private static final String EMIT_EMPTY_ROWS = "hive.test.emit.empty.rows";
+  private static final String EMIT_NUM_ROWS = "hive.test.emit.num.rows";
+
+  // Create subclass of EmbeddedThriftBinaryCLIService
+  public static class MyThriftBinaryCLIService extends EmbeddedThriftBinaryCLIService {
+    private TStatus success = new TStatus(TStatusCode.SUCCESS_STATUS);
+    private boolean emitEmptyRows;
+    private int numRows;
+    private int position;
+    private int emptyRowPos;
+
+    @Override
+    public synchronized void init(HiveConf hiveConf) {
+      hiveConf.setBoolean("hive.support.concurrency", false);
+      this.emitEmptyRows = hiveConf.getBoolean(EMIT_EMPTY_ROWS, false);
+      this.numRows = hiveConf.getInt(EMIT_NUM_ROWS, 10);
+      this.emptyRowPos = (numRows / 2);
+      this.position = 0;
+      super.init(hiveConf);
+    }
+
+    @Override
+    public TFetchResultsResp FetchResults(TFetchResultsReq req) throws TException {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
Thanks for the comments, @ellieMayVelasquez! The method is defined the thrift, and used for many other languages: php, python, etc. It's may difficult to change the method.

Reply:
noticed.